Consul.Test/CatalogTest.cs

```diff
@@ -140,6 +146,58 @@ public async Task Catalog_RegistrationDeregistration()
             Assert.Null(node.Response);
         }
 
+        [Fact]
+        public async Task Catalog_GetTaggedAddressesService()
+        {
+            var svcID = KVTest.GenerateTestKeyName();
+            var registration = new CatalogRegistration
+            {
+                Datacenter = "dc1",
+                Node = "foobar",
+                Address = "192.168.10.10",
+                Service = new AgentService
+                {
+                    ID = svcID,
+                    Service = "redis",
+                    Tags = new[] { "master", "v1" },
+                    Port = 8000,
+                    TaggedAddresses = new Dictionary<string, ServiceTaggedAddress>
+                    {
+                        {"lan", new ServiceTaggedAddress {Address = "127.0.0.1", Port = 80}},
+                        {"wan", new ServiceTaggedAddress {Address = "192.168.10.10", Port = 8000}}
+                    }
+                }
+            };
+
+            await _client.Catalog.Register(registration);
+
+            var services = await _client.Catalog.Service("redis");
+
+            var dereg = new CatalogDeregistration
+            {
+                Datacenter = "dc1",
+                Node = "foobar",
+                Address = "192.168.10.10",
+                CheckID = "service:" + svcID
+            };
+
+            await _client.Catalog.Deregister(dereg);
+
+            dereg = new CatalogDeregistration
+            {
+                Datacenter = "dc1",
+                Node = "foobar",
+                Address = "192.168.10.10"
+            };
+
+            await _client.Catalog.Deregister(dereg);
```

Comment:
I think that deregistration is not essential for this test. Should we remove it entirely?

Reply:
Noticed that too but it's been done that way for the other tests so this PR does stick to the existing style. Fine either way

Reply:
I think it is done this way only in the `Catalog_RegistrationDeregistration` which makes sense. I believe it is better to remove it from this newly added tests if it is not necessary.

Reply:
done

Reply:
@evgslyusar 
I see that You removed extra code form `Catalog_RegistrationDeregistration` method (https://github.com/G-Research/consuldotnet/pull/111/commits/aafad6f715a6b1f10b376c1e348ac0e3292d233a). I was more talking about lines 171..188 in the `Catalog_GetTaggedAddressesService` method. Can you remvoe them as well ?